@here/olp-sdk-dataservice-read/lib/client/VolatileLayerClient.ts

```diff
@@ -267,15 +271,39 @@ export class VolatileLayerClient {
             );
         }
 
+        const cache = new MetadataCacheRepository(this.settings.cache);
+
+        if (request.getFetchOption() !== FetchOptions.OnlineOnly) {
+            const partitions = cache.get(
+                request,
+                this.hrn.toString(),
+                this.layerId
+            );
+            if (partitions) {
+                return Promise.resolve({ partitions });
+            }
+        }
+
         const metaRequestBilder = await this.getRequestBuilder(
             "metadata",
             HRN.fromString(this.hrn)
         ).catch(error => Promise.reject(error));
-        return MetadataApi.getPartitions(metaRequestBilder, {
+
+        const metadata = await MetadataApi.getPartitions(metaRequestBilder, {
             layerId: this.layerId,
             additionalFields: request.getAdditionalFields(),
             billingTag: request.getBillingTag()
-        });
+        }).catch(error => Promise.reject(error));
+
+        if (request.getFetchOption() !== FetchOptions.OnlineOnly) {
+            cache.put(
+                request,
+                this.hrn.toString(),
+                this.layerId,
+                metadata.partitions
```

Comment:
Is there any chance that metadata.partitions doesn't exist or is empty?

Reply:
no